version.py

```diff
@@ -27,14 +27,69 @@
 """
 import os
 import re
+import argparse
+import logging
+import subprocess
 
 # current version
 # We use the version of the incoming release for code
 # that is under development
 __version__ = "0.8.dev0"
+# most recent tag, used for git describe validation
+__most_recent_tag__ = "v0.7.0"
+
+
+def py_str(cstr):
+    return cstr.decode("utf-8")
+
+
+def git_describe_version():
+    """Get the suffix by git describe.
+
+    Returns
+    -------
+    pub_ver: str
+        Public version.
+
+    local_ver: str
+        Local version(with additional label).
+    """
+    cmd = ["git", "describe", "--tags"]
+    proc = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+    (out, _) = proc.communicate()
+
+    if proc.returncode != 0:
+        msg = py_str(out)
+        if msg.find("not a git repository") != -1:
+            return __version__, __version__
+        logging.warning("git describ error: %", msg)
+        return __version__, __version__
+    describe = py_str(out).strip()
+    arr_info = describe.split("-")
+
+    if not arr_info[0].endswith(__most_recent_tag__):
+        logging.warning("%s does not match most recent tag %s", describe, __most_recent_tag__)
```

Comment:
My understanding is that in this case we'll use `__version__` as a fallback. Maybe that should be clear on the warning?